src/metabase/public_settings.clj

```diff
@@ -809,46 +809,78 @@ See [fonts](../configuring-metabase/fonts.md).")
                     ;; frontend should set this value to `true` after the modal has been shown once
                     v))))
 
-(defsetting uploads-enabled
-  (deferred-tru "Whether or not uploads are enabled")
-  :visibility :authenticated
-  :export?    true
-  :type       :boolean
-  :audit      :getter
-  :default    false)
-
 (defn- not-handling-api-request?
   []
   (nil? @api/*current-user*))
 
-(defn set-uploads-database-id!
-  "Sets the :uploads-database-id setting, with an appropriate permission check."
-  [new-id]
-  (if (or (not-handling-api-request?)
-          (mi/can-write? :model/Database new-id))
-    (setting/set-value-of-type! :integer :uploads-database-id new-id)
-    (api/throw-403)))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Deprecated uploads settings begin
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; These settings were removed in v50 and will be removed in a future release. They are kept here for backwards compatibility
```

Comment:
Is there a mechanism to remind us to do the removal, and is there a policy on which version this will happen in?

Reply:
Good question. There isn't a mechanism for deprecating and removing settings currently but I've decided to add `deprecated` metadata to the settings in [this commit](a24d0a2bf272b1d38f8adb76d216871e3d2a406c) and so we'll default to remove these in 53 following the standard deprecation cycle for driver changes (Cam does this manually). I also asked [here](https://metaboat.slack.com/archives/CKZEMT1MJ/p1716828710408369) to change the behaviour of the config file loading so that removing these shouldn't be such a drama. If we do that we can happily remove these settings by 53 and all we'll lose is the logged messages.